mathics/builtin/arithmetic.py

```diff
@@ -622,39 +625,59 @@ class DirectedInfinity(SympyFunction):
             "Indeterminate"
         ),
         "DirectedInfinity[args___] + _?NumberQ": "DirectedInfinity[args]",
-        "DirectedInfinity[0]": (
-            "Message[Infinity::indet,"
-            "  Unevaluated[DirectedInfinity[0]]];"
-            "Indeterminate"
-        ),
-        "DirectedInfinity[0.]": (
+        # Times. See if can be reinstalled in eval_Times
+        "Alternatives[0, 0.] DirectedInfinity[z___]": (
             "Message[Infinity::indet,"
-            "  Unevaluated[DirectedInfinity[0.]]];"
+            "  Unevaluated[0 DirectedInfinity[z]]];"
             "Indeterminate"
         ),
-        "DirectedInfinity[DirectedInfinity[x___]]": "DirectedInfinity[x]",
+        "a_?NumericQ * DirectedInfinity[b_]": "DirectedInfinity[a * b]",
+        "a_ DirectedInfinity[]": "DirectedInfinity[]",
+        "DirectedInfinity[a_] * DirectedInfinity[b_]": "DirectedInfinity[a * b]",
     }
 
     formats = {
         "DirectedInfinity[1]": "HoldForm[Infinity]",
-        "DirectedInfinity[-1]": "HoldForm[-Infinity]",
+        "DirectedInfinity[-1]": "PrecedenceForm[-HoldForm[Infinity], 390]",
```

Comment:
We should be getting numbers like 390 from the operator tables of Mathics scanner.

Reply:
Times has Precedence 400, this needs to have a slightly lower value to ensure the parenthesis.

Reply:
There is something then incomplete here. What comes in contact with what? Is there a value in [Robert's table](https://wltools.github.io/LanguageSpec/Specification/Syntax/OperatorTableHTML/) that is correct here?

Reply:
Maybe what is missing is better documentation of what `PrecedenceForm` means.
https://reference.wolfram.com/language/ref/PrecedenceForm.html

Consider this:
```
a DirectedInfinity[-1]  + b
```
In WMA is formatted as

```
c + b (-Infinity)
```

`PrecedenceForm` modifies the precedence of the outer `Times` in  `Times[b, PrecedenceForm[Times[-1, HoldForm[Infinity]], 390] ]`  to have a lower precedence value than the inner `Times` (which has the default precedence value of 400). Then, `-1 Infinity` is surrounded by a parenthesis, according to that precedence.
`390` and `399` have the same effect, as far as the expression is not a subexpression of another operator with (for example) precedence `395` .
For example, `Plus` has precedence `310`. So, let's consider now
```
 DirectedInfinity[3+4 I] + DirectedInfinity[-1]
```

This expression is formatted as
``` 
-Infinity + (3 / 5 + 4 / 5 I) Infinity
```
Notice that now there are not any parenthesis sorrounding `DirectedInfinity`. If instead of `390` we had used `290`,
then it would be formatted as

``` 
(-Infinity) + ((3 / 5 + 4 / 5 I) Infinity)
```


Now, in WL there is an operator with precedence `390`: `Star`. If we try with it (in WMA):

```
In[1]:= Star[b, DirectedInfinity[-1]]                                                                                                                                        

Out[1]= b ⋆ -Infinity
```
we see that it does not add parenthesis, so the precendece value for `Star` is not higher that `390`. Notice that if `Star` is understood as the convolution product, this is not OK, but is what WL does.

Reply:
This is good and important information. Give me some time to study this and compare with the existing tables and Robert's tables.

Reply:
Notice also that Robert's tables have several values depending on the context. I guess that  this is related to the use of PrecedenceForm in some internal  the format rules in WMA.

Reply:
Given that this is a much larger issue, and I think it is not going to be solved in the way proposed above, let's remove this change.  I created https://github.com/Mathics3/mathics-core/issues/811 to capture the discussion above, and track this problem.